packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,291 @@
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets suffixInsets;
+
+  /// The [suffixIcon] property sets the suffix widget's icon. Defaults to
+  /// the X-Mark [CupertinoIcons.xmark_circle_fill]. The suffix is customizable
+  /// so that users can override it with other options, like a bookmark icon.
+  final Icon suffixIcon;
+
+  /// The [suffixMode] property dictates when the X-Mark (suffix) should be
+  /// visible, a.k.a. only on when editing, not editing, on always, or on never.
+  /// This defaults to only on when editing.
+  final OverlayVisibilityMode suffixMode;
+
+  /// The [onSuffixTap] property lets you customize the X-Mark (suffix) action.
+  /// This defaults to clearing the text. The suffix action is customizable
+  /// so that users can override it with other functionality, that isn't
+  /// necessarily clearing text.
+  final VoidCallback? onSuffixTap;
+}
+
+class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField> {
+  @override
+  Widget build(BuildContext context) {
+    // The icon size will be scaled by a factor of the accessibility text scale,
+    // to follow the behavior of `UISearchTextField`.
+    final double scaledIconSize =
+        MediaQuery.textScaleFactorOf(context) * widget.itemSize;
+
+    // If decoration was not provided, create a decoration with the provided
+    // background color and border radius (which both have default values).
+    final BoxDecoration decoration = widget.decoration ??
+        BoxDecoration(
+          color: CupertinoDynamicColor.resolve(widget.backgroundColor, context),
+          borderRadius: widget.borderRadius,
+        );
+
+    final TextStyle placeholderStyle = widget.placeholderStyle ??
+        TextStyle(
+          color: CupertinoDynamicColor.resolve(
+              CupertinoColors.secondaryLabel, context),
+        );
+
+    final Color? itemColor =
+        CupertinoDynamicColor.resolve(widget.itemColor, context);
+
+    final IconThemeData iconThemeData =
+        IconThemeData(color: itemColor, size: scaledIconSize);
+
+    final Widget prefix = Padding(
+      child: IconTheme(
+          child: const Icon(CupertinoIcons.search), data: iconThemeData),
+      padding: widget.prefixInsets,
+    );
+
+    // TODO(DanielEdrisian): Replace [GestureDetector] with a [CupertinoButton].
+    // The reason why I went with [GestureDetector] was because
+    // [CupertinoButton] was messing with the content size of the entire
+    // search field. Must find a way to get it working with a button, so that
+    // touch-down has that fading effect.
+    final Widget suffix = Padding(
+      child: GestureDetector(
+        child: IconTheme(child: widget.suffixIcon, data: iconThemeData),
+        onTap: widget.onSuffixTap ??
+            () {
+              widget.controller!.text = '';
+              if (widget.onChanged != null) {
```

Comment:
:+1:

Reply:
nit: if the text field is already empty then `onChanged` probably shouldn't be called. Also you can just say `widget.onChanged?.call('');`